docs/content/en/docs/reference/clusterspec/irsa.md

```diff
@@ -0,0 +1,117 @@
+---
+title: "IAM for Pods configuration"
+linkTitle: "IAM for pods"
+weight: 20
+description: >
+  EKS Anywhere cluster spec for Pod IAM (IRSA)
+---
+
+### IAM Role for Service Account on EKS Anywhere clusters with self-hosted signing keys
+
+IAM Role for Service Account (IRSA) enables applications running in clusters to authenticate with AWS services using IAM roles. The current solution for leveraging this in EKS Anywhere involves creating your own OIDC provider for the cluster, and hosting your cluster's public service account signing key. The public keys along with the OIDC discovery document should be hosted somewhere that AWS STS can discover it. The steps below assume the keys will be hosted on a publicly accessible S3 bucket. Refer [this](https://docs.aws.amazon.com/AmazonS3/latest/userguide/configuring-block-public-access-bucket.html) doc to ensure that the s3 bucket is publicly accessible.
+
+The steps below are based on the [guide for configuring IRSA for DIY Kubernetes](https://github.com/aws/amazon-eks-pod-identity-webhook/blob/master/SELF_HOSTED_SETUP.md), with modifications specific to EKS Anywhere's cluster provisioning workflow. The main modification is the process of generating the keys.json document. As per the original guide, the user has to create the service account signing keys, and then use that to create the keys.json document prior to cluster creation. This order is reversed for EKS Anywhere clusters, so you will create the cluster first, and then retrieve the service account signing key generated by the cluster, and use it to create the keys.json document. The sections below show how to do this in detail.
+
+#### Create an OIDC provider and make its discovery document publicly accessible
+
+1. Create an s3 bucket to host the public signing keys and OIDC discovery document for your cluster as per [this section](https://github.com/aws/amazon-eks-pod-identity-webhook/blob/master/SELF_HOSTED_SETUP.md#create-an-s3-bucket). Ensure you follow all the steps and save the `$HOSTNAME` and `$ISSUER_HOSTPATH`.
+
+1. Create the OIDC discovery document as follows:
+
+```bash
```

Comment:
Indent each of the code blocks or sub-lists within numbered lists three spaces. Otherwise the next number will return to 1.